src/profile.js

```diff
@@ -23,45 +25,50 @@ import {withRouter} from 'react-router-dom'
 
     handleUpdate = (event) => {
       event.preventDefault();
-      if(this.completedInput()) {
-        const id = this.props?.location?.state?.id;
-        console.log(id);
-        const myForm = new FormData (document.getElementById("profileForm"));
-        myForm.append("id", id);
-        const myRequest = new Request("http://127.0.0.1:5000/profile", {
-          method: "POST",
-          body: myForm,
-        });
-
-        fetch(myRequest)
-        .then((res) =>
-          res.json())
-        .then((res) => {
-          if(res.response === "Success"){
-            this.props.history.push({
-              pathname: "/questionnaire",
-              state: {id: id}
-            });
-          }
+      if (this.checkAge()) {
+        if (this.completedInput()){
+
+          const id = this.props?.location?.state?.id;
+          console.log(id);
+          const myForm = new FormData (document.getElementById("profileForm"));
+          myForm.append("id", id);
+          const myRequest = new Request("http://127.0.0.1:5000/profile", {
+            method: "POST",
+            body: myForm,
+          });
+
+          fetch(myRequest)
+          .then((res) =>
+            res.json())
+          .then((res) => {
+            if(res.response === "Success"){
+              this.props.history.push({
+                pathname: "/questionnaire",
+                state: {id: id}
+              });
+            }
             
-          else {
+            else {
+              this.setState({
+                error: res.response,
+              });
+            }  
+          })
+          .catch((error) => {
             this.setState({
-              error: res.response,
+              error: "Error connecting to backend",
             });
-          }  
-        })
-        .catch((error) => {
-          this.setState({
-            error: "Error connecting to backend",
           });
-        });
+        }
+        else 
+          alert("Please fill in all fields")
       }
       else 
-        alert("Please fill in all fields");
+        alert("Please enter valid birthday");
```

Comment:
Do you need this here? seems like you are already alerting with reasons before you return false from function checkAge()